awx/main/scheduler/task_manager.py

```diff
@@ -521,28 +529,35 @@ def process_pending_tasks(self, pending_tasks):
                 if settings.IS_K8S and task.capacity_type == 'execution':
                     logger.debug("Skipping group {}, task cannot run on control plane".format(rampart_group.name))
                     continue
-
-                remaining_capacity = self.get_remaining_capacity(rampart_group.name, capacity_type=task.capacity_type)
-                if task.task_impact > 0 and remaining_capacity <= 0:
-                    logger.debug("Skipping group {}, remaining_capacity {} <= 0".format(rampart_group.name, remaining_capacity))
-                    continue
-
+                # at this point we know the instance group is NOT a container group
+                # because if it was, it would have started the task and broke out of the loop.
                 execution_instance = InstanceGroup.fit_task_to_most_remaining_capacity_instance(
                     task, self.graph[rampart_group.name]['instances']
                 ) or InstanceGroup.find_largest_idle_instance(self.graph[rampart_group.name]['instances'], capacity_type=task.capacity_type)
 
-                if execution_instance or rampart_group.is_container_group:
-                    if not rampart_group.is_container_group:
-                        execution_instance.remaining_capacity = max(0, execution_instance.remaining_capacity - task.task_impact)
-                        execution_instance.jobs_running += 1
-                        logger.debug(
-                            "Starting {} in group {} instance {} (remaining_capacity={})".format(
-                                task.log_format, rampart_group.name, execution_instance.hostname, remaining_capacity
-                            )
+                if execution_instance:
+                    task.execution_node = execution_instance.hostname
+                    # If our execution instance is a hybrid, prefer to do control tasks there as well.
+                    if (
+                        execution_instance.node_type == 'hybrid'
+                        and execution_instance.remaining_capacity >= task.task_impact + settings.AWX_CONTROL_PLANE_TASK_IMPACT
```

Comment:
I think `and execution_instance.remaining_capacity >= task.task_impact + settings.AWX_CONTROL_PLANE_TASK_IMPACT` should be deleted.

Let me know if my comments haven't been making sense. My proposition is that control tasks (like project updates) and the extra penalty for hybrid nodes can fairly easily be handled in `fit_task_to_most_remaining_capacity_instance`, and I have example code for it in https://github.com/ansible/awx/compare/devel...AlanCoding:control_node_fast?expand=1#diff-6b5e2004df5844765d0ce3cb503debb4fb980d19fd436c4d9516b49a6685fd06

Reply:
you addressed this in the patch I pushed